src/distilabel/steps/task/evol_instruct/generator.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+from distilabel.steps.base import RuntimeParameter
+
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+if sys.version_info < (3, 11):
+    from enum import EnumMeta as EnumType
+else:
+    from enum import EnumType
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+import numpy as np
+from pydantic import Field, PrivateAttr
+from typing_extensions import override
+
+from distilabel.steps.task.base import GeneratorTask
+from distilabel.steps.task.evol_instruct.utils import GenerationMutationTemplates
+from distilabel.steps.task.typing import ChatType
+
+if TYPE_CHECKING:
+    from distilabel.steps.typing import GeneratorStepOutput
+
+
+class EvolInstructGenerator(GeneratorTask):
+    """WizardLM: Empowering Large Language Models to Follow Complex Instructions
+
+    Reference:
+        - https://arxiv.org/abs/2304.12244
+        - https://github.com/h2oai/h2o-wizardlm
+    """
+
+    num_instructions: int
```

Comment:
Didn't experience that myself either, but sure! Also note that the `seed_texts` will always contain N instructions to evolve, where N is equal to the `num_instructions` to generate, which IMO is suboptimal in some scenarios, but overall alleviates the issue with infinite non-evolvable instructions. Anyway, worth having a look, is that somewhere in the paper and/or their implementation? Maybe I just missed it!

Reply:
No it is not but normally I would expect it to run x generations, where x is an input argument, like with the non-generator. In my example when I ran the pipeline it took 4 generations to evolve an evolution from scratch, so I thought it might have a risk of racking up un-expected costs/run time for some people.